Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM python:3-slim-buster as base
+
+ENV user_name="app-user"
+
+# requires python 3.x
+# image includes Python 3.11.3
+
+# requires poetry
+RUN pip3 install poetry
+
+# requires /opt/todoapp with appropriate permissions for app user
+RUN mkdir -p /opt/todoapp/todo_app
```

Comment:
This can be unnecessary if we tweak our copy commands to make it clear what's a folder by using a trailing slash, (also combined in case that's of interest):
```
COPY poetry.lock pyproject.toml /opt/todoapp/
```